src/main/java/net/snowflake/ingest/streaming/internal/ChannelContext.java

```diff
@@ -0,0 +1,108 @@
+package net.snowflake.ingest.streaming.internal;
+
```

Comment:
Personally I like this class as it is immutable and it summarises the "id" of the channel. This makes is a nice candidate for "sharing" information with other classes like the `channelData`. 

Actually why not passing the context as is to the `flush()` method as argument so that we can pass it to the constructor of the `ChannelData` directly, instead of having the `channelData#setChannelContext()`? I think this makes it cleaner. WDYT @sfc-gh-azagrebin ?

Reply:
True, I think the builder pattern also makes sense in a separate PR to make the channelData immutable